cmd/util/ledger/migrations/storage_v6.go

```diff
@@ -1055,15 +1061,15 @@ func (c *ValueConverter) VisitStringValue(_ *oldInter.Interpreter, value *oldInt
 func (c *ValueConverter) VisitArrayValue(_ *oldInter.Interpreter, value *oldInter.ArrayValue) bool {
 	newElements := make([]newInter.Value, 0)
 
+	arrayStaticType := ConvertStaticType(value.StaticType()).(newInter.ArrayStaticType)
+
 	for _, element := range value.Elements() {
-		newElement := c.Convert(element)
+		newElement := c.Convert(element, arrayStaticType.ElementType())
```

Comment:
Maybe refactor `arrayStaticType.ElementType()` out of the loop into a variable, as it is effectively constant

```suggestion
		newElement := c.Convert(element, arrayStaticType.ElementType())
```